### PR TITLE
TST: Add tests to ensure proper conversion of datetime64/timedelta64/timestamp/duration to and from pyarrow

### PR DESCRIPTION
**Pull request by Julian048**
- [ ] closes #54191 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by Julian048** [comment]
As per [this issue from the pyarrow repository](https://github.com/apache/arrow/issues/33321) there was error in convert_dtypes and pa.table()/pd.to_pandas() in which pyarrow would always return [ns] as unit regardless of associated metadeta. I have added tests to ensure the conversion always preserves associated unit. 

Pyarrow has since fixed and closed the issue in their current main branch (dev version 13.0), and so when it is publicly available the if conditionals mentioned in #54191 can be considered for removal with these tests to ensure functionality remains the same after their removal.

**Review by mroeschke — CHANGES_REQUESTED** [review]
Can you only add 1 test for the example in https://github.com/pandas-dev/pandas/issues/54191? The test should go in `test_convert_dtypes.py`

**Comment by Julian048** [comment]
I've fixed the test to match the issue and put it in the proper location, but I'm still having a failing CI "minimum versions" and I can't pinpoint why if anyone could help with that.

**Comment by mroeschke** [comment]
Might help to merge main again

File: pandas/tests/frame/methods/test_convert_dtypes.py [hunk 1]
```diff
@@ -167,3 +167,18 @@ def test_convert_dtypes_pyarrow_to_np_nullable(self):
         result = ser.convert_dtypes(dtype_backend="numpy_nullable")
         expected = pd.DataFrame(range(2), dtype="Int32")
         tm.assert_frame_equal(result, expected)
+
+    def test_convert_dtypes_timestamp_and_duration(self):
+        # GH 54191
+        pytest.importorskip("pyarrow")
+        timestamp_series = pd.Series(
```
Comment (mroeschke) [inline pandas/tests/frame/methods/test_convert_dtypes.py:174]: Can you mirror the exact code snippet in https://github.com/pandas-dev/pandas/issues/54191#issue-1811847359?

Comment (Julian048) [reply pandas/tests/frame/methods/test_convert_dtypes.py:174]: This is what you're referencing right?
s = pd.Series(pd.date_range("2020-01-01", "2020-01-02", freq="1min"))
s = s.astype("timestamp[ms][pyarrow]")
t = s.convert_dtypes(dtype_backend="pyarrow")
pd.testing.assert_series_equal(s, t)

Comment (mroeschke) [reply pandas/tests/frame/methods/test_convert_dtypes.py:174]: Correct

**Comment by Julian048** [comment]
I apologize for all the commits, git is new to me

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Julian048

**Pull request merged**